internal/service/certificatesigningrequest.go

```diff
@@ -110,10 +110,22 @@ func (h *ServiceHandler) ListCertificateSigningRequests(ctx context.Context, req
 func (h *ServiceHandler) CreateCertificateSigningRequest(ctx context.Context, request server.CreateCertificateSigningRequestRequestObject) (server.CreateCertificateSigningRequestResponseObject, error) {
 	orgId := store.NullOrgId
 
-	// don't set fields that are managed by the service
-	request.Body.Status = nil
 	common.NilOutManagedObjectMetaProperties(&request.Body.Metadata)
 
+	signedCert, err := signApprovedCertificateSigningRequest(h.ca, *request.Body)
+	if err != nil {
+		return nil, err
+	}
+	approvedCondition := api.Condition{
+		Type:    api.CertificateSigningRequestApproved,
+		Status:  api.ConditionStatusTrue,
+		Reason:  "Approved",
+		Message: "Approved",
```

Comment:
Would it make sense to provide the signal that this is an `AutoApproved` CSR? Maybe that is not important at this point.

Reply:
Yes, but in a different way. 
This should be recorded in "who approved". If we do not have "who approved" our approval data structure needs fixing.
Let me have a look at what do we have there.